core/src/main/java9/com/linecorp/armeria/internal/Java9VersionSpecific.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implementation of {@link JavaVersionSpecific} using Java 9 APIs.
+ */
+class Java9VersionSpecific extends Java8VersionSpecific {
```

Comment:
I wonder extending `Java8VersionSpecific` is good for here.
How about extending `JavaVersionSpecific` and delegating `Java8VersionSpecific` if you need?

Reply:
I think it's very common for newer versions to still use most of the APIs from previous ones, and override just one or two, so inheriting makes that simpler compared to always delegating everything in newer classes. For Java 9, it's not so useful since it's the first level of overriding, but I'd expect Java 10 to benefit from inheriting from Java 9. I figured Java 9 could inherit from 8 for consistency, but what do you think?

Reply:
This makes me wonder if it's a good idea to merge Java8VersionSpecific into JavaVersionSpecific. WDYT?

Reply:
Thanks for the kind explanation. The reason why I avoid inheritance is I think some [APIs could be removed](https://dzone.com/articles/apis-to-be-removed-from-java-11) in next versions of Java. But I agree with your opinion.

Reply:
Good idea - merged 8 implementation into `JavaVersionSpecific` since it is indeed the "defualt implementation"

Reply:
I have a question on dependency between multiple java versions.
```
main/java
    /java9
    /java10 
    /java1x
```
I'm not special on this. But I want to check this case. If some application work on Java 10, does it combine `java` + `java9` + `java10`?
I think it is possible to see only `java` + `java10` which do not include `java9`.

I will pull this PR and check it. :-)

Reply:
Thanks @ikhoon. Let us know once the result is out.